frame/staking/src/pallet/impls.rs

```diff
@@ -599,20 +599,19 @@ impl<T: Config> Pallet<T> {
 
 	/// Apply previously-unapplied slashes on the beginning of a new era, after a delay.
 	fn apply_unapplied_slashes(active_era: EraIndex) {
-		let slash_defer_duration = T::SlashDeferDuration::get();
-		<Self as Store>::EarliestUnappliedSlash::mutate(|earliest| {
-			if let Some(ref mut earliest) = earliest {
-				let keep_from = active_era.saturating_sub(slash_defer_duration);
-				for era in (*earliest)..keep_from {
-					let era_slashes = <Self as Store>::UnappliedSlashes::take(&era);
-					for slash in era_slashes {
-						slashing::apply_slash::<T>(slash, era);
-					}
-				}
-
-				*earliest = (*earliest).max(keep_from)
+		if <Self as Store>::UnappliedSlashes::contains_key(&active_era) {
+			let era_slashes = <Self as Store>::UnappliedSlashes::take(&active_era);
```

Comment:
can't we just `<Self as Store>::UnappliedSlashes::take` without asking storage about entry existence?

Reply:
My wrong intention here was to prevent the need to read a potentially read a big storage item, but I am wrong and this makes no difference -- if it does not exist, `::take` and `containst_key` are the same, else, we have to read it anyhow.